src/backend/access/transam/xlog.c

```diff
@@ -748,7 +771,16 @@ typedef struct XLogCtlData
 	 * XLOG_FPW_CHANGE record that instructs full_page_writes is disabled.
 	 */
 	XLogRecPtr	lastFpwDisableRecPtr;
-	XLogRecPtr  lastWrittenPageLSN;
+
+	/*
+	 * Maximal last written LSN for pges not present in lastWrittenLsnCache
+	 */
+	XLogRecPtr  maxLastWrittenLsn;
+
+	/*
+	 * Double linked list to implement LRU replacement policy for last written LSN cache
+	 */
+	LastWrittenLsnCacheEntry lastWrittenLsnLRU;
```

Comment:
It seems that these variables are protected by `LastWrittenLsnLock` as the lock is acquired at the start of `GetLastWrittenLSN` and `SetLastWrittenLSN`. Can we add a comment to describe this protection?